documentation/AuthServer.md

```diff
@@ -34,7 +35,9 @@ The document types used by the authorization server include:
 | `name` | String | The client name. | None |
 | `oauth2ClientId` | String | The unique identifier issued to a registered OAuth2 client by the authorization server. | None |
 | `oauth2ClientSecret` | String | Confidential shared secret between the client and the authorization server, used to verify the identity of the client. | None |
-| `redirectUri` | String | The URL on client's site where users will be sent after authorization. | None |
+| `oauth2ClientType` | String | The type of clients that use different OAuth2 grant type. GPII supports 4 type of clients: <br>`nativeApp`: native GPII apps that use OAuth2 resource owner password credentials grant.<br>`webApp`: web applications.<br>`thirdPartyApp`: third party applications that GPII users can define privacy settings for. <br>`clientCredentialsApp`: applications that use OAuth2 client credentials grant to add new preferences on behalf of users, such as First Discovery Tool.| Must be one of these values: "nativeApp", "webApp", "thirdPartyApp", "clientCredentialsApp".|

```

Comment:
The `oauth2ClientType` here is feeling unclear to me. I think we could rework this taxonomy to make it clearer. Some issues that I am seeing with the current organization:

1. The name `oauth2ClientType` suggests to me that the values would be OAUTH2 grant types but only one appears to be a grant type: `clientCredentialsApp`
1. We have `nativeApp` and `webApp` -- is `thirdPartyApp` neither a native app nor a web app?
1. My feeling is that the term `nativeApp` is too general

I'm thinking that we could either switch to a taxonomy of OAUTH2 grant types or alternatively a taxonomy of types of applications within the GPII ecosystem. We have a bit of a mix of both at the moment.